### PR TITLE
Code text optimization process tx

### PR DESCRIPTION
**Pull request by mche**
No logic changes.

Code text changes only for processing any response tx of API.

New `'oauth2.process_tx'` helper usefull for processing user profile response, for example.

### CODE REVIEW

**Comment by jhthorsen** [comment]
Please squash the commits and force push to this PR. The gitlog is very confusing.

https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/

File: lib/Mojolicious/Plugin/OAuth2.pm [hunk 1]
```diff
@@ -4,9 +4,9 @@ use Mojo::Base 'Mojolicious::Plugin';
 use Mojo::UserAgent;
 use Mojo::Util 'deprecated';
 use Carp 'croak';
-use strict;
+#~ use strict;
 
-our $VERSION = '1.53';
+our $VERSION = '1.5399';
```
Comment (jhthorsen) [inline lib/Mojolicious/Plugin/OAuth2.pm]: Don't change the version number. I will do that on a new release.

File: lib/Mojolicious/Plugin/OAuth2.pm [hunk 2]
```diff
@@ -4,9 +4,9 @@ use Mojo::Base 'Mojolicious::Plugin';
 use Mojo::UserAgent;
 use Mojo::Util 'deprecated';
 use Carp 'croak';
-use strict;
+#~ use strict;
```
Comment (jhthorsen) [inline lib/Mojolicious/Plugin/OAuth2.pm]: Just remove this line.

File: lib/Mojolicious/Plugin/OAuth2.pm [hunk 3]
```diff
@@ -40,18 +40,11 @@ sub register {
...
-    }
-    else {
-      $providers->{$provider} = $config->{$provider};
-    }
+  while ( my ($name, $vals) = each %$config ) {
+    @{ $providers->{$name} ||= {} }{ keys %$vals } = values %$vals;
   }
 
-  $self->providers($providers);
+  #~ $self->providers($providers); hashref not need
```
Comment (jhthorsen) [inline lib/Mojolicious/Plugin/OAuth2.pm]: Just remove this line.

**Comment by mche** [comment]
Ok, Jan. There isnt pod about new helper. Please add few pod lines or later.

**Comment by jhthorsen** [comment]
@marcusramberg: Please decide.

**Comment by mche** [comment]
Ok, please check new pod lines

**Comment by marcusramberg** [comment]
Guess I wasn't able to see the utility of this PR. Sorry.

**Pull request closed without merging**